repos/system_upgrade/el7toel8/actors/readopensshconfig/libraries/readopensshconfig.py

```diff
@@ -0,0 +1,93 @@
+import os
+import errno
+
+from leapp.libraries.stdlib import CalledProcessError, run
+from leapp.models import OpenSshConfig, OpenSshPermitRootLogin
+from leapp.libraries.stdlib import api
+
+
+def line_empty(s):
+    return len(s) == 0 or s.startswith('\n') or s.startswith('#')
+
+
+def parse_config(config):
+    ''' Parse OpenSSH server configuration or the output of sshd test option '''
+
+    # RHEL7 defaults
+    r = OpenSshConfig(
+        permit_root_login=[],
+    )
+
+    in_match = None
+    for line in config:
+        s = line.strip()
+        if line_empty(s):
+            continue
+
+        l = s.split()
+        if len(l) < 2:
+            continue
+        value = l[1]
+        if l[0].lower() == 'match':
+            in_match = l[1:]
+            continue
+
+        if l[0].lower() == 'permitrootlogin':
+            # convert deprecated alias
+            if value == "without-password":
+                value = "prohibit-password"
+            v = OpenSshPermitRootLogin(value=value, in_match=in_match)
+            r.permit_root_login.append(v)
+
+        elif l[0].lower() == 'useprivilegeseparation':
+            # Record only first occurence, which is effective
+            if not r.use_privilege_separation:
+                r.use_privilege_separation = value
+
+        elif l[0].lower() == 'protocol':
+            # Record only first occurence, which is effective
+            if not r.protocol:
+                r.protocol = value
+
+        elif l[0].lower() == 'ciphers':
+            # Record only first occurence, which is effective
+            if not r.ciphers:
+                r.ciphers = value
+
+        elif l[0].lower() == 'macs':
+            # Record only first occurence, which is effective
+            if not r.macs:
+                r.macs = value
+
+    return r
+
+def produce_config(producer, config):
+    ''' Produce a Leapp message with all interesting OpenSSH configuration
+        options '''
+
+    producer(config)
+
+
+def read_sshd_config():
+    ''' Read the actual sshd configuration file '''
+    try:
+        with open('/etc/ssh/sshd_config', 'r') as f:
+            return f.readlines()
+    except IOError as e:
+        if e.errno != errno.ENOENT:
+            api.current_logger().warn('Failed to read sshd_config: %s' % e)
+        return []
+
+
+def scan_sshd(producer):
+    ''' Parse sshd_config configuration file to create OpenSshConfig message.
```

Comment:
Malformed docstring - use double quotes, closing quotes are on the same line as the opening ones for one-line docstring, see the [guidelines](https://github.com/oamg/leapp-repository/blob/master/CONTRIBUTING.md#code-style).